test/IntegrationSpec.hs

```diff
@@ -122,47 +111,55 @@ spec = do
-                analysisFunc = Clojure.discover,
-                analysisProjects = [[reldir|.|]]
-              }
-          ]
-      }
+  let lein = filteredLein (const True)
+      filteredLein :: (Leiningen.LeiningenProject -> Bool) -> Path Rel Dir -> [Path Rel Dir] -> Spec
+      filteredLein sieve root paths = repo
+        Repo
+          { repoRoot = root,
+            repoPrebuildScript = Nothing,
+            repoDynamicNixDeps = ["leiningen"],
+            repoAnalyses =
+              [ Analysis
+                  { analysisName = "leiningen",
+                    analysisFinder = fmap (filter sieve) . Leiningen.findProjects,
+                    analysisFunc = Leiningen.getDeps,
+                    analysisMkProject = Leiningen.mkProject,
+                    analysisProjects = map simpleTestProject paths
+                  }
+              ]
+          }
+  -- FIXME: Unsupported major Java version. Expects 8 or 11.\n{:major \"15\", :minor \"0\"}
+  pending $ lein [reldir|repos/clojure/puppetserver|] [[reldir|.|]]
 
-  repo
-    Repo
-      { repoRoot = [reldir|repos/clojure/ring|],
-        repoPrebuildScript = Nothing,
-        repoDynamicNixDeps = ["leiningen"],
-        repoAnalyses =
-          [ Analysis
-              { analysisName = "Clojure",
-                analysisFunc = Clojure.discover,
-                analysisProjects = [[reldir|.|]]
-              }
-          ]
-      }
+  lein [reldir|repos/clojure/ring|]
+    [ [reldir|./|],
+      [reldir|ring-bench/|],
+      [reldir|checkouts/ring-core/|],
+      [reldir|checkouts/ring-devel/|],
+      [reldir|checkouts/ring-devel/checkouts/ring-core/|],
+      [reldir|checkouts/ring-jetty-adapter/|],
+      [reldir|checkouts/ring-jetty-adapter/checkouts/ring-core/|],
+      [reldir|checkouts/ring-jetty-adapter/checkouts/ring-servlet/|],
+      [reldir|checkouts/ring-jetty-adapter/checkouts/ring-servlet/checkouts/ring-core/|],
+      [reldir|checkouts/ring-servlet/|],
+      [reldir|checkouts/ring-servlet/checkouts/ring-core/|],
+      [reldir|ring-core/|],
+      [reldir|ring-devel/|],
+      [reldir|ring-devel/checkouts/ring-core/|],
+      [reldir|ring-jetty-adapter/|],
+      [reldir|ring-jetty-adapter/checkouts/ring-core/|],
+      [reldir|ring-jetty-adapter/checkouts/ring-servlet/|],
+      [reldir|ring-jetty-adapter/checkouts/ring-servlet/checkouts/ring-core/|],
+      [reldir|ring-servlet/|],
+      [reldir|ring-servlet/checkouts/ring-core/|]
+    ]
 
-  repo
-    Repo
-      { repoRoot = [reldir|repos/clojure/eastwood|],
-        repoPrebuildScript = Nothing,
-        repoDynamicNixDeps = ["leiningen"],
-        repoAnalyses =
-          [ Analysis
-              { analysisName = "Clojure",
-                analysisFunc = Clojure.discover,
-                analysisProjects = [[reldir|.|]]
-              }
-          ]
-      }
+  let noCrucible Leiningen.LeiningenProject {..} = not . T.isInfixOf "crucible" . T.pack $ toFilePath leinDir
```

Comment:
if only the `leinDir` field is being used here, we can make this point-free: `noCrucible = not . T.isInfixOf "crucible" . T.pack . toFilePath . leinDir`